src/main/kotlin/screeps/utils/Options.kt

```diff
@@ -1,7 +1,23 @@
 package screeps.utils
 
+import screeps.api.CreepMemory
+import screeps.api.DirectionConstant
 import screeps.api.Game
+import screeps.api.SpawnEnergyProvider
+import screeps.api.structures.StructureSpawn
 
 fun routeOptions(routeCallback: (roomName: String, fromRoomName: String) -> Double): Game.Map.RouteOptions = jsObject {
     this.routeCallback = routeCallback
 }
+
+fun spawnOwptions(
```

Comment:
why not make this a class?

`class SpawnOptions(override val memory: CreepMemory?) : SpawnOptions`

Reply:
Classes require metadata to be made and has other overhead that is not necessary for options.
The reason Javascript functions take objects is to get a sort of named arguments. Kotlin has this already and it can be done with a function.

Reply:
We discuss this in another PR 👍